### PR TITLE
Add entities, perms, and API handlers for approvals.

### PR DESCRIPTION
**Pull request by jrobbins**
In this CL:
+ Model class for approvals
+ API handlers to get and set approvals
+ Permission checking to see who can set approvals
+ Fetching and parsing the API_OWNERS file
+ Added known approvals to the various processes.

### CODE REVIEW

**Comment by jrobbins** [comment]
The Travis CI failures are due to an SSL error in the CI service.  I don't think that they are related to this PR, and this code passes all unit tests locally.

File: internals/approval_defs.py
Comment (jstenback) [inline internals/approval_defs.py:74]: Guessing you don't want to leave this print() in...

**Review by jstenback — APPROVED** [review]
Approved, but please have a look at the comments I left.

File: internals/approval_defs.py
Comment (jstenback) [inline internals/approval_defs.py:91]: Did you mean to set afd.approvers here as well? If not, I don't see how it's ever not a string?

Comment (jrobbins) [reply internals/approval_defs.py:91]: That's correct.  In the only examples that we have right now for API Owners the value of afd.approvers will be a URL to fetch.  We keep that URL so that we can fetch it again periodically.    In the future, we may have some other approval types that just have a hard-coded list of approvers.  I'll add a comment to explain that.

**Comment by cwilso** [comment]
Is this for API owner LGTMs, or horizontal reviews like Security/Privacy?

**Comment by jrobbins** [comment]
> Is this for API owner LGTMs, or horizontal reviews like Security/Privacy?

API owners at first, but I am laying the foundation to generalize to horizontal reviews eventually.

**Review by jstenback — APPROVED** [review]

**Pull request merged**
